packages/jest-matchers/src/matchers.js

```diff
@@ -408,6 +408,76 @@ const matchers: MatchersObject = {
   },
 
   toMatchSnapshot,
+
+  toMatchObject(receivedObject: Object, expectedObject: Object) {
+
+    if (typeof receivedObject !== 'object' || receivedObject === null) {
+      throw new Error(
+        matcherHint('[.not].toMatch', 'object', 'expected') + '\n\n' +
+        `${RECEIVED_COLOR('string')} value must be an object.\n` +
```

Comment:
Whoops, thanks for catching that, haha a bit embarrassing that I didn't see this;) 

I implemented your suggestion:

```
<green>expected</green> value must be an object. Received:
```